packages/archway-proto/tsconfig.json

```
@@ -0,0 +1,20 @@
+{
+    "compilerOptions": {
+        "declaration": true,
+        "declarationDir": "./types",
+        "emitDeclarationOnly": true,
+        "isolatedModules": true,
+        "esModuleInterop": true,
+        "allowJs": true,
+        "lib": [
+            "es2015",
+            "DOM"
+        ]
+    },
+    "include": [
+        "src/**/*"
+    ],
+    "exclude": [
+        "node_modules"
+    ]
+}
```

Can you please try building it with this setup instead?

```suggestion
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "build",
    "rootDir": "src",
    "emitDeclarationOnly": true,
    "isolatedModules": true,
    "allowJs": true,
  },
  "include": [
    "src"
  ],
  "exclude": [
    "src/**/*.spec.ts",
    "src/**/__mocks__",
  ]
}
```

---

This setting change causes an error on the extern.ts gen file. Some of the functions are returning sometimes and sometimes not based on logic.

davidchoi@Davids-MacBook-Pro arch3-proto % yarn run arch:ts
src/codegen/extern.ts:12:34 - error TS7030: Not all code paths return a value.

12 export const getRpcEndpointKey = (rpcEndpoint: string | HttpEndpoint) => {
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/codegen/extern.ts:23:15 - error TS7030: Not all code paths return a value.

23     if (!key) return;
                 ~~~~~~~
Found 2 errors in the same file, starting at: src/codegen/extern.ts:12

---

I took your config and added only one extra compilerOption field: "noImplicitReturns": false

This compiles. Is this an acceptable compromise?

---

Looks like the declaration and declarationDir options are also needed to export the d.ts files.